packages/platform/sdk/src/lib/endpoints.ts

```diff
@@ -9,19 +9,19 @@ type BaseEndpointDeclaration = {
 // Define a type for static URIs
 type StaticUriEndpoint = BaseEndpointDeclaration & {
   uri: string;
-  constructUri?: never; // Ensure constructUri is not used here
+  constructUriFromObject?: never;
+  constructUriFromArray?: never;
 };
 
 // Introduce a generic type for dynamic URIs, allowing for explicit parameter type declaration
-type DynamicUriEndpoint<ParamsType extends Record<string, string> | string[]> = BaseEndpointDeclaration & {
+type DynamicUriEndpoint = BaseEndpointDeclaration & {
   uri?: never; // Ensure uri is not used here
-  constructUri: (params: ParamsType) => string;
+  constructUriFromArray?: (params: string[]) => string;
+  constructUriFromObject?: (params: Record<string, string>) => string;
 };
 
 // Create a discriminated union of the two types, incorporating generics
-type EndpointDeclaration<ParamsType extends Record<string, string> | string[] = string[]> =
-  | StaticUriEndpoint
-  | DynamicUriEndpoint<ParamsType>;
+type EndpointDeclaration = StaticUriEndpoint | DynamicUriEndpoint;
```

Comment:
// Define a type for to avoid verbosity
`type StrArrOrRecordStrStr = string[] | Record<string, string>;`